### PR TITLE
BAU: Test RP generate it's own tokens

### PR DESCRIPTION
**Pull request by andy-paine**
An amalgamation of some code @willp-bl wrote to use JWT to validate
tokens in test RP and some stuff Aditya and I wrote to generate tokens
using test RP (using the private key)

This will allow us to stop running token services and just rely on the
test RP instances to do auth themselves as per [ADR #11](https://github.com/alphagov/verify-architecture/blob/master/adr/0011-removal-of-token-service.md)

Supercedes #25

### CODE REVIEW

File: src/main/java/uk/gov/ida/rp/testrp/Urls.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
     String SUCCESSFUL_REGISTER_PATH = "/success";
     String LOGOUT_PATH = "/logout";
     String SUCCESSFUL_IDP_PATH = "/success-idp" ;
+    String GENERATE_TOKEN_PATH = "/generate-token";
```
[the remark this answers is not in the document]
Comment (adityapahuja) [reply src/main/java/uk/gov/ida/rp/testrp/Urls.java]: This has been done.

[the remark this answers is not in the document]
Comment (willp-bl) [reply src/main/java/uk/gov/ida/rp/testrp/Urls.java]: great :)

File: README.md [hunk 2]
```diff
@@ -6,6 +6,19 @@ A stub relying party used when testing the GOV.UK Verify hub.
 
 * This service is only for internal testing and is not representative of how a service should connect to Verify. For an authorative implementation of how a service can connect to Verify please see the [Technical Onboarding Guide](https://alphagov.github.io/rp-onboarding-tech-docs/) and the [Verify Service Provider](https://www.github.com/alphagov/verify-service-provider).
 
+## Tokens
+
+Test RP can use JWT to control access if the `privateBetaUserAccessRestrictionEnabled` flag is set to true. The following request can be used to generate these certificates.
+
+```HTTP
+POST /generate-token HTTP/1.1
```
Comment (willp-bl) [inline README.md]: this is wrong but it can wait

Comment (adityapahuja) [reply README.md]: This has been fixed.

**Pull request merged**